index.js

```diff
@@ -1,310 +1,333 @@
-    console.log("Signature Missing.")
-    return
+    console.log("Signature is missing.");
+    return;
+  }
+
+  let stringToVerify;
+
+  if (req.method === "POST") {
+    stringToVerify = req.body.data.toString();
+  } else if (req.method === "GET") {
+    stringToVerify = req._parsedUrl.query;
+  }
+
+  let computedSignature = crypto
+    .createHmac("sha256", process.env.CLIENT_SECRET)
+    .update(stringToVerify)
+    .digest("hex");
+
+  if (req.headers["x-asana-request-signature"] !== computedSignature) {
+    console.log("Request cannot be verified.");
+    return;
+  } else {
+    console.log("Request verified!");
   }
 
-  // TODO: Replace signature check
+  next();
+});
 
-  // let stringToVerify;
-  // if (req.method === "POST") {
-  //   stringToVerify = req.body.toString();
-  // } else if (req.method === "GET") {
-  //   stringToVerify = req.url.query;
-  // }
-  // let computedSignature = require("crypto").createHmac("sha256", secret)
-  //   .update(stringToVerify)
-  //   .digest("hex");
-  // if (req.headers["x-asana-request-signature"] !== computedSignature) {
-  //   return;
-  // }
+// Client endpoints
+app.get("/", (req, res) => {
+  res.send("Hello World!");
+});
 
-  next()
-})
+app.get("/auth", (req, res) => {
+  // We recommend creating a secure Oauth flow (https://developers.asana.com/docs/oauth)
+  console.log("Auth happened!");
+  res.sendFile(path.join(__dirname, "/auth.html"));
+});
 
-app.get('/widget', (req, res) => {
-  console.log("Widget Happened!")
-  res.json(widget_response)
-})
+// API endpoints
+app.get("/widget", (req, res) => {
+  console.log("Widget happened!");
+  res.json(widget_response);
+});
 
-app.get('/form/metadata', (req, res) => {
-  console.log("Form Happened!")
-  res.json(form_response)
-})
+app.get("/form/metadata", (req, res) => {
+  console.log("Modal Form happened!");
+  res.json(form_response);
+});
 
-app.get('/search/typeahead', (req, res) => {
-  console.log("Typeahead Happened!")
-  res.json(typeahead_response)
-})
+app.get("/search/typeahead", (req, res) => {
+  console.log("Typeahead happened!");
+  res.json(typeahead_response);
+});
 
-app.post('/form/onchange', (req, res) => {
-  console.log("OnChange Happened!")
-  console.log(req.body)
-  res.json(form_response)
-})
+app.post("/form/onchange", (req, res) => {
+  console.log("OnChange happened!");
+  console.log(req.body);
+  res.json(form_response);
+});
 
-app.post('/search/attach', (req, res) => {
-  console.log("Attach Happened!")
-  console.log(req.body)
-  res.json(attachment_response)
-})
+app.post("/search/attach", (req, res) => {
+  console.log("Attach happened!");
+  console.log(req.body);
+  res.json(attachment_response);
+});
 
-app.post('/form/submit', (req, res) => {
-  console.log("Modal Submitted!")
-  console.log(req.body)
-  res.json(attachment_response)
-})
+app.post("/form/submit", (req, res) => {
+  console.log("Modal Form submitted!");
+  console.log(req.body);
+  res.json(attachment_response);
+});
```

Comment:
Try: would be super valuable to include a comment linking to the API reference for each of these endpoints.